engine/src/main/java/com/datatorrent/stram/StramClient.java

```diff
@@ -114,7 +116,6 @@
   private final int amPriority = 0;
   private ApplicationId appId;
   private final LogicalPlan dag;
-  public String javaCmd = "${JAVA_HOME}" + "/bin/java";
```

Comment:
Although not part of API, making this private could break someone's code in case they were using it (I am not aware of any)

Reply:
All engine classes are not subject of semantic versioning and they can't be used outside of the core itself.